src/effects/backends/builtin/beatgrindeffect.cpp

```diff
@@ -0,0 +1,192 @@
+#include "effects/backends/builtin/beatgrindeffect.h"
+
+#include <QtDebug>
+
+#include "util/sample.h"
+#include "util/math.h"
+#include "util/rampingvalue.h"
+
+
+// static
+QString BeatGrindEffect::getId() {
+    return "org.mixxx.effects.beatgrind";
+}
+
+// static
+EffectManifestPointer BeatGrindEffect::getManifest() {
+    EffectManifestPointer pManifest(new EffectManifest());
+
+    pManifest->setAddDryToWet(true);
+    pManifest->setEffectRampsFromDry(true);
+
+    pManifest->setId(getId());
+    pManifest->setName(QObject::tr("BeatGrind"));
+    pManifest->setShortName(QObject::tr("BeatGrind"));
+    pManifest->setAuthor("The Mixxx Team");
+    pManifest->setVersion("1.0");
+    pManifest->setDescription(QObject::tr(
+      "Stores the input signal in a temporary buffer and loops it"));
+    pManifest->setMetaknobDefault(db2ratio(-3.0));
+
+    EffectManifestParameterPointer delay = pManifest->addParameter();
+    delay->setId("loop_length");
+    delay->setName(QObject::tr("Length"));
+    delay->setShortName(QObject::tr("Length"));
+    delay->setDescription(QObject::tr(
+        "Length of the loop\n"
+        "1/8 - 2 beats if tempo is detected\n"
+        "1/8 - 2 seconds if no tempo is detected"));
+    delay->setValueScaler(EffectManifestParameter::ValueScaler::LINEAR);
+    delay->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
+    delay->setUnitsHint(EffectManifestParameter::UnitsHint::BEATS);
+    delay->setRange(0.0, 0.5, 2.0);
+
+    EffectManifestParameterPointer quantize = pManifest->addParameter();
+    quantize->setId("quantize");
+    quantize->setName(QObject::tr("Quantize"));
+    quantize->setShortName(QObject::tr("Quantize"));
+    quantize->setDescription(QObject::tr(
+        "Round the Time parameter to the nearest 1/4 beat."));
+    quantize->setValueScaler(EffectManifestParameter::ValueScaler::TOGGLE);
+    quantize->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
+    quantize->setUnitsHint(EffectManifestParameter::UnitsHint::UNKNOWN);
+    quantize->setRange(0, 1, 1);
```

Comment:
Is there a use case for not quantizing this effect?

Reply:
I can't think of any, remove?